src/main/resources/templates/pet/edit.html

```diff
@@ -64,48 +60,66 @@
                             <input type="datetime-local" id="birthDate" name="birthDate" th:field="*{birthDate}">
                             <label th:if="${#fields.hasErrors('birthDate')}" th:errors="*{birthDate}"></label>
                             <br/>
-                            <label for="dewormed"><h4 th:text="#{pet.dewormed}"/></label>
-                            <input type="checkbox" name='dewormed' th:field="*{dewormed}" class="form-control"
-                                   placeholder="dewormed" id='dewormed'/>
+                            <label for="dewormed"><h4 th:text="#{pet.dewormed}"></h4></label>
+                            <input type="checkbox" name="dewormed" th:field="*{dewormed}" class="form-control"
+                                   id="dewormed"/>
                             <label th:if="${#fields.hasErrors('dewormed')}" th:errors="*{dewormed}"></label>
                             <br/>
-                            <label for="sterilized"><h4 th:text="#{pet.sterilized}"/></label>
-                            <input type="checkbox" name='sterilized' th:field="*{sterilized}" class="form-control"
-                                   placeholder="sterilized" id='sterilized'/>
+                            <label for="sterilized"><h4 th:text="#{pet.sterilized}"></h4></label>
+                            <input type="checkbox" name="sterilized" th:field="*{sterilized}" class="form-control"
+                                   id="sterilized"/>
                             <label th:if="${#fields.hasErrors('sterilized')}" th:errors="*{sterilized}"></label>
                             <br/>
-                            <label for="vaccinated"><h4 th:text="#{pet.vaccinated}"/></label>
-                            <input type="checkbox" name='vaccinated' th:field="*{vaccinated}" class="form-control"
-                                   placeholder="vaccinated" id='vaccinated'/>
-                            <label th:if="${#fields.hasErrors('vaccinated')}" th:errors="*{vaccinated}"></label>
-                            <br/>
+                            <!-- Vaccination Records Table -->
+                            <label><h4 th:text="#{pet.vaccinationRecords}"></h4></label>
+                            <table class="table table-bordered">
+                                <thead>
+                                    <tr>
+                                        <th>Vaccination</th>
+                                        <th>Status</th>
+                                    </tr>
+                                </thead>
+                                <tbody>
+                                    <tr th:each="vaccination : ${petCommand.vaccinations}">
+                                        <td th:text="${vaccination.name}"></td>
+                                        <td>
+                                            <select th:field="*{vaccinations[__${vaccination.index}__].status}">
```

Comment:
You need to read the current vaccines status value here, I do not think this code will do the job.